src/codeu/chat/server/Controller.java

```diff
@@ -46,18 +46,16 @@ public Controller(Uuid serverId, Model model, File persistentPath) {
     this.uuidGenerator = new RandomUuidGenerator(serverId, System.currentTimeMillis());
     this.persistentPath = persistentPath;
 
-    TransactionLog.restore(persistentPath, this);
-    LOG.info("restored called");
+    PersistenceLog.restore(this, persistentPath);
   }
 
   @Override
   public Message newMessage(Uuid author, Uuid conversation, String body) {
     Uuid id = createId();
     Time time = Time.now();
 
-    Object[] params = new Object[]{id, body, time.inMs(), conversation, author};
     checkBuffer();
-    TransactionLog.writeLog("message", params);
+    PersistenceLog.writeTransaction("message", id, body, time.inMs(), author, conversation);
```

Comment:
re. transaction types, you have manually typed them here ("message", "admin" etc) and you have these as constants in PersistenceLog as well. So if ever you want to change a transaction type, you need to do it in both places, and if you ever forget to change them here, the entire persistence system fails. To avoid that, make those constants public and use them here. PersistenceLog.MESSAGE, PersistenceLog.ADD_USER and so on...